src/Core/MySQL/MySQLUtils.cpp

```diff
@@ -13,7 +13,7 @@ namespace MySQLUtils
 {
 
 DecimalUtils::DecimalComponents<DateTime64>
-getNormalizedDateTime64Components(const DataTypePtr data_type, const ColumnPtr col, const int row_num)
+getNormalizedDateTime64Components(DataTypePtr data_type, ColumnPtr col, size_t row_num)
```

Comment:
Just curious: was `const` misused here somehow?

Reply:
Nah. More a matter of taste. Some people like const value parameters because it will make the compiler complain:

```cpp
void func(const int i)
{
    i = 3;  // compiler complains here 
};
```

The thing is that `i` is passed by value and changes to it don't propagate outside of `func`. Some people (including me) like this more:

```cpp
void func(int i)
{
    i = 3;
    for (int j = 0; j < i; ...) ... // use i like it was a regular variable
};
```

The same argument can be made for the `*Ptr` arguments. In ClickHouse, the non-const form is more popular.